tftypes/value_json.go

```diff
@@ -16,7 +16,25 @@ import (
 // terraform-plugin-go.  Third parties should not use it, and its behavior is
 // not covered under the API compatibility guarantees. Don't use this.
 func ValueFromJSON(data []byte, typ Type) (Value, error) {
-	return jsonUnmarshal(data, typ, NewAttributePath())
+	return jsonUnmarshal(data, typ, NewAttributePath(), ValueFromJSONOpts{})
+}
+
+// ValueFromJSONOpts contains options that can be used to modify the behaviour when
+// unmarshalling JSON.
+//
+// IgnoreUndefinedAttributes is used to ignore any attributes which appear in the
+// JSON but do not have a corresponding entry in the schema. For example, raw state
+// where an attribute has been removed from the schema.
+type ValueFromJSONOpts struct {
+	IgnoreUndefinedAttributes bool
```

Comment:
Nit: If you move the field documentation above the attribute, Go docs will show that text when linked directly to the field 😄 

```suggestion
type ValueFromJSONOpts struct {
	// IgnoreUndefinedAttributes is used to ignore any attributes which appear in the
	// JSON but do not have a corresponding entry in the schema. For example, raw state
	// where an attribute has been removed from the schema.
	IgnoreUndefinedAttributes bool
```